tests/SpecTests/ClientSideEncryption/Prose22_RangeExplicitEncryptionTest.php

```diff
@@ -0,0 +1,479 @@
+ * @group csfle
+ * @group serverless
+ */
+class Prose22_RangeExplicitEncryptionTest extends FunctionalTestCase
+{
+    private $clientEncryption;
+    private $collection;
+    private $encryptedClient;
+    private $key1Id;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->isStandalone() || ($this->isShardedCluster() && ! $this->isShardedClusterUsingReplicasets())) {
+            $this->markTestSkipped('Range explicit encryption tests require replica sets');
+        }
+
+        if (version_compare($this->getServerVersion(), '7.0.0', '<')) {
+            $this->markTestSkipped('Range explicit encryption tests require MongoDB 7.0 or later');
+        }
+
+        $client = static::createTestClient();
+
+        $key1Document = $this->decodeJson(file_get_contents(__DIR__ . '/../client-side-encryption/etc/data/keys/key1-document.json'));
+        $this->key1Id = $key1Document->_id;
+
+        // Drop the key vault collection and insert key1Document with a majority write concern
+        self::insertKeyVaultData($client, [$key1Document]);
+
+        $this->clientEncryption = $client->createClientEncryption([
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)]],
+        ]);
+
+        $autoEncryptionOpts = [
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)]],
+            'bypassQueryAnalysis' => true,
+        ];
+
+        $this->encryptedClient = self::createTestClient(null, [], [
+            'autoEncryption' => $autoEncryptionOpts,
+            /* libmongocrypt caches results from listCollections. Use a new
+             * client in each test to ensure its encryptedFields is applied. */
+            'disableClientPersistence' => true,
+        ]);
+    }
+
+    public function setUpWithTypeAndRangeOpts(string $type, array $rangeOpts): void
+    {
+        if ($type === 'DecimalNoPrecision' || $type === 'DecimalPrecision') {
+            $this->markTestSkipped('Bundled libmongocrypt does not support Decimal128 (PHPC-2207)');
+        }
+
+        /* Read the encryptedFields file directly into BSON to preserve typing
+         * for 64-bit integers. This means that DropEncryptedCollection and
+         * CreateEncryptedCollection will be unable to inspect the option for
+         * metadata collection names, but that's not necessary for the test. */
+        $encryptedFields = Document::fromJSON(file_get_contents(__DIR__ . '/../client-side-encryption/etc/data/range-encryptedFields-' . $type . '.json'));
+
+        $database = $this->encryptedClient->selectDatabase($this->getDatabaseName());
+        $database->dropCollection('explicit_encryption', ['encryptedFields' => $encryptedFields]);
```

Comment:
Why do you need to set the encryptedFields when dropping the collection?

Reply:
This is mentioned in "Test Setup" section of [22. Range Explicit Encryption](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#range-explicit-encryption) but it's not explained in detail.

In addition to the main collection, queryable encryption uses additional metadata collections. The driver's `createCollection()` helper makes those when an `encryptedFields` option is provided. If an `encryptedFields` option is passed to the drop helper, it will additionally drop those metadata collections. The notable difference is that `encryptedFields` is actually an option on the `create` command, but not `drop`. The option here is entirely used client-side to tell PHPLIB to clean up additional collections. [Drop Collection Helper](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#drop-collection-helper) in the CSFLE spec has more detail on that.